apps/passport-client/src/dispatch.ts

```diff
@@ -756,7 +774,13 @@ async function doSync(
     state.pcds,
     state.subscriptions
   );
+
   if (state.serverStorageHash !== appStorage.storageHash) {
+    if (!validateAndLogStateErrors(state.self, state.identity, state.pcds)) {
```

Comment:
Why not put this inside uploadSerializedStorage and check the result?
I don't like having the validation in 2 places (here and also in uploadStorage) when it could be one.  I also dislike the asymmetry that `uploadStorage` and `uploadSerializedStorage` have different error-handling assumptions, which isn't clear from the name.

Reply:
I put it here in order to avoid having to pass down those parameters into the function. I'm fine making the change you suggest.

Reply:
Looks like uploadSerializedStorage is updated now, but this call site needs to be udpated to remove the extra validation, and instead check for the ValidationError result.

Reply:
My review raced with the change which addresses this.